--- Makefile.original	2024-07-08 15:09:27.140329081 +0200
+++ Makefile	2024-07-08 15:09:56.300989002 +0200
@@ -1114,6 +1114,11 @@
 	@echo '====  Run ./llama-cli -h for help.  ===='
 	@echo
 
+llama-cli-a: examples/main/main.cpp \
+	$(OBJ_ALL)
+	$(CXX) $(CXXFLAGS) -c $< -o $(call GET_OBJ_FILE, $<)
+	ar rcs libllama-cli.a $(filter-out %.h $<,$^) $(call GET_OBJ_FILE, $<)
+
 llama-infill: examples/infill/infill.cpp \
 	$(OBJ_ALL)
 	$(CXX) $(CXXFLAGS) -c $< -o $(call GET_OBJ_FILE, $<)
