--- llama.cpp-master/examples/llava/llava.cpp	2024-12-01 15:05:26.915691953 +0100
+++ llama.cpp/examples/llava/llava.cpp	2024-12-01 15:08:30.859787620 +0100
@@ -382,7 +382,8 @@
     return true;
 }
 
-bool llava_image_embed_make_with_clip_img(clip_ctx * ctx_clip, int n_threads, const clip_image_u8 * img, float ** image_embd_out, int * n_img_pos_out) {
+bool llava_image_embed_make_with_clip_img(clip_ctx * ctx_clip, int n_threads, const void * _img, float ** image_embd_out, int * n_img_pos_out) {
+    const clip_image_u8 * img = (const clip_image_u8 *)_img;
     int num_max_patches = 6;
     if (clip_is_minicpmv(ctx_clip)) {
         num_max_patches = 10;
