--- llama.cpp-master/examples/main/main.cpp	2024-11-25 14:16:56.890105909 +0100
+++ llama.cpp/examples/main/main.cpp	2024-11-26 11:36:48.107556828 +0100
@@ -31,6 +31,20 @@
 #pragma warning(disable: 4244 4267) // possible loss of data
 #endif
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef LLAMA_LIB
+typedef void (*_llama_yield_token_t)(const char * token);
+typedef int (*_llama_should_stop_t)(void);
+int _llama_cli_main(int argc, char ** argv, _llama_yield_token_t _llama_yield_token, _llama_should_stop_t _llama_should_stop);
+#endif
+
+#ifdef __cplusplus
+}
+#endif
+
 static llama_context           ** g_ctx;
 static llama_model             ** g_model;
 static common_sampler          ** g_smpl;
@@ -91,7 +105,15 @@
     return formatted;
 }
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef LLAMA_LIB
+int _llama_cli_main(int argc, char ** argv, _llama_yield_token_t _llama_yield_token, _llama_should_stop_t _llama_should_stop) {
+#else
 int main(int argc, char ** argv) {
+#endif
     common_params params;
     g_params = &params;
     if (!common_params_parse(argc, argv, params, LLAMA_EXAMPLE_MAIN, print_usage)) {
@@ -673,7 +695,24 @@
                 const std::string token_str = common_token_to_piece(ctx, id, params.special);
 
                 // Console/Stream Output
+#ifdef LLAMA_LIB
+                if (_llama_yield_token != NULL) {
+                    char* c_token_str = (char*)malloc(token_str.length() + 1);
+                    std::memset(c_token_str, 0, token_str.length() + 1);
+                    std::strcpy(c_token_str, token_str.c_str());
+                    _llama_yield_token(c_token_str);
+                    free((void*)c_token_str);
+                }
+
+                if (_llama_should_stop != NULL) {
+                    if (_llama_should_stop() == 1) {
+                        n_remain = 0;
+                        break;
+                    }
+                }
+#else
                 LOG("%s", token_str.c_str());
+#endif
 
                 // Record Displayed Tokens To Log
                 // Note: Generated tokens are created one by one hence this check
@@ -895,3 +934,7 @@
 
     return 0;
 }
+
+#ifdef __cplusplus
+}
+#endif
