--- ../Makefile	2024-07-10 22:17:00.507999852 +0200
+++ Makefile	2024-07-10 22:40:13.002343655 +0200
@@ -425,6 +425,31 @@
 	MK_CXXFLAGS += -pg
 endif
 
+#
+# llama-cpp-cffi
+# Set shared library extension and linker flags based on the platform
+#
+ifeq ($(UNAME_S), Linux)
+    LIB_EXT := so
+    LIB_LDFLAGS := -shared
+    LIB_CXXFLAGS := -fPIC -DLLAMA_LIB
+endif
+
+ifeq ($(UNAME_S), Darwin)
+    LIB_EXT := dylib
+    LIB_LDFLAGS := -dynamiclib
+    LIB_CXXFLAGS := -fPIC -DLLAMA_LIB
+endif
+
+# For Windows (assuming MinGW)
+ifeq ($(OS), Windows_NT)
+    LIB_EXT := dll
+    LIB_LDFLAGS := -shared
+    LIB_CXXFLAGS := -DLLAMA_LIB
+endif
+
+LIB_NAME := llama_cli.$(LIB_EXT)
+
 # Architecture specific
 # TODO: probably these flags need to be tweaked on some architectures
 #       feel free to update the Makefile for your architecture and send a pull request or issue
@@ -1132,6 +1157,16 @@
 	@echo '====  Run ./llama-cli -h for help.  ===='
 	@echo
 
+llama-cli-shared: examples/main/main.cpp \
+	$(OBJ_ALL)
+	$(CXX) $(CXXFLAGS) $(LIB_CXXFLAGS) -c $< -o $(call GET_OBJ_FILE, $<)
+	$(CXX) $(CXXFLAGS) $(LIB_CXXFLAGS) $(filter-out %.h $<,$^) $(call GET_OBJ_FILE, $<) -o $(LIB_NAME) $(LIB_LDFLAGS) $(LDFLAGS)
+
+llama-cli-static: examples/main/main.cpp \
+	$(OBJ_ALL)
+	$(CXX) $(CXXFLAGS) $(LIB_CXXFLAGS) -c $< -o $(call GET_OBJ_FILE, $<)
+	ar rcs llama_cli.a $(filter-out %.h $<,$^) $(call GET_OBJ_FILE, $<)
+
 llama-infill: examples/infill/infill.cpp \
 	$(OBJ_ALL)
 	$(CXX) $(CXXFLAGS) -c $< -o $(call GET_OBJ_FILE, $<)
