--- llama.cpp-master/Makefile	2024-12-04 20:47:44.525404966 +0100
+++ llama.cpp/Makefile	2024-12-05 10:27:02.954736735 +0100
@@ -980,6 +980,9 @@
 	$(DIR_COMMON)/build-info.o \
 	$(DIR_COMMON)/json-schema-to-grammar.o

+OBJ_LLAVA = \
+
+
 OBJ_ALL = $(OBJ_GGML) $(OBJ_LLAMA) $(OBJ_COMMON)

 LIB_GGML   = $(LIB_PRE)ggml$(DSO_EXT)
@@ -1385,10 +1388,15 @@
 	examples/llava/llava.h \
 	examples/llava/clip.cpp \
 	examples/llava/clip.h \
+	examples/llava/mllama.cpp \
+	examples/llava/mllama.h \
 	common/stb_image.h \
 	common/base64.hpp \
 	$(OBJ_ALL)
-	$(CXX) $(CXXFLAGS) -static -fPIC -c $< -o $@ -Wno-cast-qual
+	$(CXX) $(CXXFLAGS) -c examples/llava/llava.cpp -o examples/llava/llava.o
+	$(CXX) $(CXXFLAGS) -c examples/llava/clip.cpp -o examples/llava/clip.o
+	$(CXX) $(CXXFLAGS) -c examples/llava/mllama.cpp -o examples/llava/mllama.o
+	ar rcs $@ examples/llava/llava.o examples/llava/clip.o examples/llava/mllama.o

 llama-llava-cli: examples/llava/llava-cli.cpp \
 	examples/llava/llava.cpp \
